frame/ibc/ibc-primitives/src/lib.rs

```diff
@@ -28,11 +28,22 @@ pub struct SendPacketData {
 	/// This value should be in nano seconds
 	pub timeout_timestamp: u64,
 	/// port id as utf8 string bytes
-	pub port_id: Vec<u8>,
+	pub port_id: PortId,
 	/// channel id as utf8 string bytes
-	pub channel_id: Vec<u8>,
-}
-#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Eq, Ord, PartialOrd)]
+	pub channel_id: ChannelId,
+}
+#[derive(
+	codec::Encode,
+	Default,
+	Debug,
+	codec::Decode,
+	Clone,
+	PartialEq,
+	Eq,
+	Ord,
+	PartialOrd,
+	scale_info::TypeInfo,
```

Comment:
what's this doing?

Reply:
https://paritytech.github.io/substrate/master/frame_support/pallet_prelude/trait.TypeInfo.html